hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -934,9 +929,17 @@ public boolean isBucketEmpty(String volume, String bucket)
 
           // Entry should not be marked for delete, consider only those
           // entries.
-          if(!deletedKeySet.contains(kv.getKey())) {
-            cacheKeyMap.put(kv.getKey(), kv.getValue());
-            currentCount++;
+          String key = kv.getKey();
+          try {
+            CacheKey<String> k = new CacheKey<>(key);
+            boolean inCache = keyTable.isKeyInCache(k);
+            CacheValue<OmKeyInfo> omKeyInfoCacheValue =
+                keyTable.getCacheValue(new CacheKey(k));
+            if(!inCache || omKeyInfoCacheValue != null) {
```

Comment:
I think `omKeyInfoCacheValue` is basically the same as `inCache`. It will be null if the entry is not in the cache, but I think your intention here is to check `omKeyInfoCacheValue.getCacheValue() != null` ?

Reply:
Thank you. I found it easier to add a new explicit API but will revert this part of the code and just use the null checks. :-)